src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java

```diff
@@ -22,33 +22,59 @@
 import com.vmware.vipclient.i18n.messages.dto.MessagesDTO;
 import com.vmware.vipclient.i18n.util.ConstantsKeys;
 import com.vmware.vipclient.i18n.util.JSONUtils;
+import com.vmware.vipclient.i18n.util.LocaleUtility;
 
 public class StringService {
     Logger              logger = LoggerFactory.getLogger(StringService.class);
     
     @SuppressWarnings("unchecked")
     public String getString(MessagesDTO dto) {
     	String key = dto.getKey();
+    	
+    	MessageCacheItem cacheItem = getComponentLocaleCache(dto);
+    	
+    	// If failed to get MessageCacheItem of a non-default locale, 
+    	// use MessageCacheItem of the default locale instead. 
+    	Map<String, String> cacheOfComponent = cacheItem.getCachedData();	
+    	if (cacheOfComponent.isEmpty() && !LocaleUtility.isDefaultLocale(dto.getLocale())) {
+			MessagesDTO defaultLocaleDTO = new MessagesDTO(dto.getComponent(), 
```

Comment:
For the first call, the cacheOfComponent will be empty and then fallback to default value? is this correct?

Reply:
Yes, this is correct. If query for the target locale failed, then query for the default locale (which may or may not be the same as the source locale).  See the workflow design:

https://confluence.eng.vmware.com/display/GQ/Singleton+Client+Library+-+Caching%2C+Fallback+Mechanism%2C+and+Other+Features#SingletonClientLibrary-Caching,FallbackMechanism,andOtherFeatures-L10nQuery